process.c

```diff
@@ -8507,8 +8587,11 @@ InitVM_process(void)
     rb_define_method(rb_cWaiter, "pid", detach_process_pid, 0);
 
     rb_cProcessStatus = rb_define_class_under(rb_mProcess, "Status", rb_cObject);
+    rb_define_alloc_func(rb_cProcessStatus, rb_process_status_allocate);
```

Comment:
Doesn't this make it possible to `Process::Status.allocate`?
I think that should be avoided since ``Process::Status` is immutable and only created by the VM.

Reply:
`Process::Status.allocate` is already part of the test suite, so it's out of scope for this PR.

Reply:
Indeed.
Why is this change needed, what is it doing then?

Reply:
Because I changed it to used typed data.